telos-frontend/src/components/todo/DashboardTodo.jsx

```diff
@@ -1,3 +1,376 @@
+import MenuItem from '@material-ui/core/MenuItem';
+import DeleteIcon from '@material-ui/icons/Delete';
+import BlockIcon from '@material-ui/icons/Block';
+import ScheduleIcon from '@material-ui/icons/Schedule';
+import Button from '@material-ui/core/Button';
+import './DashboardTodo.css';
 
-export default DashboardToDo;
+const useStyles = makeStyles((theme) => ({
+  root: {
+    width: '100%',
+    maxWidth: 360,
+    backgroundColor: theme.palette.background.paper,
+  },
+  datecontainer: {
+    display: 'flex',
+    flexWrap: 'wrap',
+  },
+  datetextField: {
+    marginLeft: theme.spacing(1),
+    marginRight: theme.spacing(1),
+    width: '100%',
+  },
+}));
+
+const outdated = {
+  color: 'red',
+};
+
+const DashboardTodo = () => {
+  let listitems = [
+    {
+      name: 'OnGoing',
+      due: '2021-04-30',
+      onGoing: true,
+      completed: false,
+    },
+    {
+      name: 'OutDated',
+      due: '2021-01-30',
+      onGoing: false,
+      completed: true,
+    },
+    {
+      name: '一二三四五',
+      due: '2021-04-30',
+      onGoing: true,
+      completed: true,
+    },
+    {
+      name: '上山打老虎',
+      due: '2021-04-30',
+      onGoing: true,
+      completed: false,
+    },
+  ];
+
+  const classes = useStyles();
+  const [item, setItem] = useState('');
+  const [dueDate, setDueDate] = useState('2021-01-01');
+  const [itemList, setItemList] = useState(listitems);
+  const [newItem, setNewItem] = useState(itemList);
+  const [anchorEl, setAnchorEl] = useState(null);
+  const [cancel, setCancel] = useState([0]);
+  const [reDate, setReDate] = useState('');
+  const [reEventName, setReEventName] = useState('');
+  const [reItem, setReItem] = useState('');
+
+  const reDateChange = (event) => {
+    setReDate(event.target.value);
+    setReItem({ name: reItem.name, due: event.target.value, onGoing: true, completed: false });
+  };
+
+  const handleToggle = (value) => () => {
+    let newList = [...newItem];
+    for (let x of newList) {
+      if (x.name === value.name) {
+        x.completed = !x.completed;
+      }
+    }
+
+    setNewItem(newList);
+    setItemList(newList);
+  };
+
+  const firstEvent = (event) => {
+    setItem({ name: event.target.value, due: dueDate, onGoing: true, completed: false });
+  };
+
+  const dateChange = (event) => {
+    setDueDate(event.target.value);
+    setItem({ name: item.name, due: event.target.value, onGoing: true, completed: false });
+  };
+
+  const secondEvent = () => {
+    setNewItem((prev) => [...prev, item]);
+    setItem('');
+  };
+
+  const handleOption = (value) => (event) => {
+    setReEventName(value);
+    // reEventName = value;
+    setReItem({ name: value, due: reDate, onGoing: true, completed: false });
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handleOptionClose = (event) => {
+    setAnchorEl(null);
+  };
+
+  const inputAlert = () => {
+    alert('Please enter something :)');
+  };
+
+  const cancelevent = (value) => () => {
```

Comment:
These functions to change the data should have associated unit tests to check that data has actually been changed.